src/server.c

```diff
@@ -3598,17 +3591,10 @@ void rejectCommandFormat(client *c, const char *fmt, ...) {
 /* This is called after a command in call, we can do some maintenance job in it. */
 void afterCommand(client *c) {
     UNUSED(c);
-    if (!server.in_nested_call) {
-        /* If we are at the top-most call() we can propagate what we accumulated.
-         * Should be done before trackingHandlePendingKeyInvalidations so that we
-         * reply to client before invalidating cache (makes more sense) */
-        if (server.core_propagates) {
-            postExecutionUnitOperations();
-        }
-        /* Flush pending invalidation messages only when we are not in nested call.
-         * So the messages are not interleaved with transaction response. */
-        trackingHandlePendingKeyInvalidations();
```

Comment:
it looks as if it's no longer relevant to avoid doing that invalidations in when nested (you moved the `if` inside).
maybe it's wrong (to check server.execution_nesting in postExecutionUnitOperations, trackingHandlePendingKeyInvalidations, etc)?

maybe it's better that all callers explicitly check it and understand when it takes action and when not (keep that function only doing actions without restrictions).
rather than making it look as if you call it and it does something when in fact it's just a "maybe".

Reply:
but on the other hand, postExecutionUnitOperations should never be called if execution_nesting is >0, so i can move the `if` outside, but then i'll add an assertion

same for trackingHandlePendingKeyInvalidations

Reply:
i didn't delete any comments, i moved them (one is in postExecutionUnitOperations and the other is in trackingHandlePendingKeyInvalidations)